spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/response/MultipartContentOperationPreprocessor.java

```diff
@@ -0,0 +1,44 @@
+package capital.scalable.restdocs.response;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.restdocs.operation.OperationRequest;
+import org.springframework.restdocs.operation.OperationRequestFactory;
+import org.springframework.restdocs.operation.OperationRequestPart;
+import org.springframework.restdocs.operation.OperationRequestPartFactory;
+import org.springframework.restdocs.operation.preprocess.OperationPreprocessorAdapter;
+
+import java.util.List;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.stream.Collectors.toList;
+
+public class MultipartContentOperationPreprocessor extends OperationPreprocessorAdapter {
```

Comment:
Is there a reason why you are extending `OperationPreprocessorAdapter` instead of implementing `ContentModifier` like the `BinaryReplacementContentModifier`? Implementing `ContentModifier` makes the code a bit simpler.

Reply:
I extending OperationPreprocessorAdapter because ContentModifier takes original content (look at  [ContentModifyingOperationPreprocessor](https://github.com/spring-projects/spring-restdocs/blob/master/spring-restdocs-core/src/main/java/org/springframework/restdocs/operation/preprocess/ContentModifyingOperationPreprocessor.java). In case of multipart/form-data content is empty byte array and all binary data holds in `parts` field.

Reply:
Thanks, this makes sense.